pkg/http/sse.go

```diff
@@ -2,22 +2,28 @@ package http
 
 import (
 	"context"
+	"net/http"
+
 	"github.com/go-chi/render"
 	"github.com/pkg/errors"
-	"net/http"
 
 	"github.com/ThreeDotsLabs/watermill"
 	"github.com/ThreeDotsLabs/watermill/message"
 	"github.com/ThreeDotsLabs/watermill/pubsub/gochannel"
 )
 
 type StreamAdapter interface {
-	// GetResponse returns the response to be sent back to client.
-	// Any errors that occur should be handled and written to `w`, returning false as `ok`.
-	GetResponse(w http.ResponseWriter, r *http.Request) (response interface{}, ok bool)
-	// Validate validates if the incoming message should be handled by this handler.
-	// Typically this involves checking some kind of model ID.
-	Validate(r *http.Request, msg *message.Message) (ok bool)
+	// InitialStreamResponse returns the first stream response to be sent back to client.
+	// Any errors that occur should be handled and written to `w`.
+	// Returning `ok` equal false ends processing the HTTP request.
+	InitialStreamResponse(w http.ResponseWriter, r *http.Request) (response interface{}, ok bool)
+	// NextEvent returns the next stream response to be sent back to the client.
+	// Typically this involves checking some kind of model ID extracted from the `msg`.
+	// The response is sent to the client only if `ok` is true.
+	// Any errors that occur should be either:
+	//    1) logged and skipped, returning (nil, false)
+	//    2) sent back to the client, returning (errorStruct, true)
+	NextEvent(r *http.Request, msg *message.Message) (response interface{}, ok bool)
```

Comment:
Maybe we also do not need that http.Request? If we would have just NextEvent(msg) this would be super elegant.

Reply:
Interesting, this could work. At first I thought we need it, as we need to somehow know what entity ID is being fetched. But since the other method is always called, it could be kept in the structure to compare later.

But then we lose access to `r.Context()`, so perhaps it'd be good to pass `ctx` explicitly?

Reply:
Actually, this would mean we need to create one StreamAdapter per request, which complicates things. So I guess keeping the request is a simpler solution.

Reply:
👍